Add a proxy_requests section and various fixes/updating wording (#885)
sensu/sensu-docs: content/sensu-go/5.0/guides/monitor-external-resources.md

    parent: guides
---

- [Using a proxy entity to monitor a website](#using-a-proxy-entity-to-monitor-a-website)
- [Using proxy requests to monitor a group of websites](#using-proxy-requests-to-monitor-a-group-of-websites)

Proxy entities allow Sensu to monitor external resources
on systems or devices where a Sensu agent cannot be installed, like a
network switch or a website.

> You can create [proxy entities][1] using [sensuctl][8], the [Sensu API][9], or the [`proxy_entity_name` check attribute][2]. When executing checks that include a `proxy_entity_name`, Sensu agents report the resulting event under the proxy entity instead of the agent entity.

Reviewer: Results are also associated with proxy entities when the check specifies values for `proxy_request` attribute.

This guide requires a running Sensu backend, a running Sensu agent, and a sensuctl instance configured to connect to the backend as a user with read and create permissions for entities, checks, and events.

## Using a proxy entity to monitor a website

In this section, we'll monitor the status of [sensu.io](https://sensu.io) by configuring a check with a **proxy entity name** so that Sensu creates an entity representing the site and reports the status of the site under this entity.

### Installing an HTTP check script

First, we'll install a [bash script][4], named `http_check.sh`, to perform an HTTP
check using **curl**.

{{< highlight shell >}}

> sudo curl https://raw.githubusercontent.com/sensu/sensu-go/60e6a68aecb0c8e0c2dc51714e08462eb81b4413/examples/checks/http_check.sh \

Reviewer: We can use a shorter URL if we reference a tagged commit, e.g. `https://raw.githubusercontent.com/sensu/sensu-go/2.0.0-beta.8-1/examples/checks/http_check.sh` works

I hope engineering will be retroactively tagging GA and newer releases in the sensu-go repo soon, so we could use a non-beta tag.

Reviewer: Tags have been backfilled!

```suggestion
sudo curl https://raw.githubusercontent.com/sensu/sensu-go/5.1.0/examples/checks/http_check.sh \
```

-o /usr/local/bin/http_check.sh && \
sudo chmod +x /usr/local/bin/http_check.sh
{{< /highlight >}}

> _PRO TIP: While this command is appropriate when running a few agents, you should consider

Reviewer: ```suggestion
_PRO TIP: While this command may be appropriate when running a few agents, you should consider
```

using [Sensu assets][5] or a [configuration management][15] tool to provide

> runtime dependencies for checks in bigger environments._

Reviewer: ```suggestion
runtime dependencies._
```

### Creating the check

Now that our script is installed, we'll create a check named
`check-http`, which runs the command `http_check.sh https://sensu.io`, at an
interval of 60 seconds, for all entities subscribed to the `proxy`
subscription, using the `sensu-site` proxy entity name.

> Create a file called `check.json` and add the following check definition.

Reviewer: ```suggestion
Create a file called `check.json` with the following check definition as it's content.
```
